### ISSUE TITLE
Unnecessarily moving imports to TYPE_CHECKING

### ISSUE THREAD

**Issue by SpangleLabs** [snok/flake8-type-checking#84]
To start with, I love this plugin, avoiding these import loops has always been such a pain when adding type hints, and this automates dealing with that, which is excellent!

However, it seems unecessarily overzealous, unless I'm misunderstanding something.

If I start with just this import:
```
from typing import Optional
```
This plugin will suggest it be changed to:
```
from typing import TYPE_CHECKING

if TYPE_CHECKING:
    from typing import Optional
```

But given we're still importing `typing`, if there were a circular dependency there, it would not be solved, right?
It seems like if there is any import from a given module, it shouldn't suggest moving the other imports in that module to TYPE_CHECKING, simply because there's no point?

For a slightly more practical example, if I have a file something like this:
```
from client import Client, Message


def send_hello_world(client: "Client") -> None:
    client.send_message(Message("hello world"))
```
Then this plugin would suggest changing it to this:
```
from typing import TYPE_CHECKING

from client import Message

if TYPE_CHECKING:
    from client import Client


def send_hello_world(client: "Client") -> None:
    client.send_message(Message("hello world"))
```

But if we're already importing client.py outside the `if TYPE_CHECKING` block, if there's a cyclic dependency, it's gonna bite us whether or not the `Client` import is wrapped in the if statement?

Am I missing something, or is the plugin just a bit overzealous?

**Comment by sondrelg** [comment]
Yeah this is a good point. I guess I've just initially gone with the approach of flagging all imports (individually) that are only used for type hints, and therefore are not needed at runtime. 

I agree though, that when an import already exists to the same file, there probably isn't a benefit to guarding the second (unless you just really like the organisation of imports into two categories), so perhaps we should change the default behaviour to take other imports into consideration. The old behaviour could be enabled with a `strict` flag or something like that I guess.

It might be worth noting that we have a few categories of imports: 
- Application imports (same module or app)
- Library imports
- Built-in library imports
- Futures imports

Import circularity is only an issue in the first category - do you think it would still make sense to change this behaviour for other categories? 

I've always wanted to know exactly what performance implications imports actually have on a large application, and whether there is an additional cost to additional imports from the same module. If there is one, I guess it would make sense to keep the current behaviour, while if there isn't, we should change this for all types. Does that make sense to you?

**Comment by SpangleLabs** [comment]
That makes some sense, yeah. I would be surprised if there's any performance hit to it, as the vast majority of the loading time would be in parsing and loading up the module, and assigning a couple more things from the module will just be variable assignments effectively? 

Would it help to actually do some timeit testing of a trivial example? It seems to be like it would be a marginal saving, if any. Maybe some memory saving

**Comment by sondrelg** [comment]
I've tried looking into it a little, and get the impression that it's not worth worrying about. I think your suggestion applied to all types of imports makes sense 👍 I'll see if I can make some progress on this today 🙂

**Comment by sondrelg** [comment]
Hello again @SpangleLabs 🙂 The plugin (in the main branch) should now behave as you suggested, and I'm more or less ready to review the changes and release v2. 

But before I do, would you mind testing whether it works for you, and/or if you notice any weirdness? You should be able to install it with `pip install git+https://github.com/snok/flake8-type-checking` and run it with `flake8 .`. Running the main branch via pre-commit is a little trickier I think.

**Comment by SpangleLabs** [comment]
Oooh!
(I had to use `poetry add --dev git+https://github.com/snok/flake8-type-checking.git#main` to install it btw. Poetry's error messages for not finding )

I tried running it in two projects, one which had lots of recomendations which I hadn't yet followed, and looks like it cut that list down a lot! 
And tried it on a project I've already adjusted to follow the recommendations, and it doesn't suggest changing them all back, which is also great!

Looks fantastic to me, thank you so much for this plugin, and thank you for updating that so fast!
